### PR TITLE
doc: Improve distance_fade documentation

### LINKED ISSUE #36051: Mismatch between Distance Fade properties and their documentation
In the documentation, it states:

    float distance_fade_max_distance

Distance at which the object fades fully and is no longer visible.

    float distance_fade_min_distance

Distance at which the object starts to fade. If the object is less than this distance away it will appear normal.

In reality, it's the other way around. The object is invisible at min and fully visible at max.

### PR DESCRIPTION
**Pull request by jslcontributor**
Changed the documentation of distance_fade_max_distance and distance_fade_min_distance to be consistent with the behavior of the code as well as the user documentation.  

Also noted the behavior of swapping max_distance and min_distance (where max_distance < min_distance).   

Fixes godotengine#36051

### CODE REVIEW

File: doc/classes/SpatialMaterial.xml
Comment (clayjohn) [inline doc/classes/SpatialMaterial.xml]: It looks like you are using spaces instead of tabs. For proper formatting of the .xml files, you need to use tabs.

Comment (clayjohn) [inline doc/classes/SpatialMaterial.xml]: "at normal opaqueness" -> "fully opaque"

File: doc/classes/SpatialMaterial.xml
Comment (clayjohn) [inline doc/classes/SpatialMaterial.xml]: change to "[b]Note:[/b] If [code]distance_fade_max_distance[/code] is less than [code]distance_fade_min_distance[/code], the behavior will be reversed, the object will start to fade away at [code]distance_fade_max_distance[/code] and fully disappear once it reaches [code]distance_fade_min_distance[/code].

File: doc/classes/SpatialMaterial.xml
Comment (clayjohn) [inline doc/classes/SpatialMaterial.xml]: "less transparent" -> "visible"

File: doc/classes/SpatialMaterial.xml
Comment (clayjohn) [inline doc/classes/SpatialMaterial.xml]: change to "[b]Note:[/b] If [code]distance_fade_min_distance[/code] is greater than [code]distance_fade_max_distance[/code], the behavior will be reversed, the object will start to fade away at [code]distance_fade_max_distance[/code] and fully disappear once it reaches [code]distance_fade_min_distance[/code].

**Review by clayjohn — DISMISSED** [review]
Looks good so far! Just a few things to fix up before it is ready to merge. :)

**Comment by akien-mga** [comment]
Rebased and applied @clayjohn's improvements.

File: doc/classes/BaseMaterial3D.xml
Comment (Calinou) [inline doc/classes/BaseMaterial3D.xml:168]:
```suggestion
			[b]Note:[/b] If [code]distance_fade_max_distance[/code] is less than [code]distance_fade_min_distance[/code], the behavior will be reversed. The object will start to fade away at [code]distance_fade_max_distance[/code] and will fully disappear once it reaches [code]distance_fade_min_distance[/code].
```

File: doc/classes/BaseMaterial3D.xml
Comment (Calinou) [inline doc/classes/BaseMaterial3D.xml:172]:
```suggestion
			[b]Note:[/b] If [code]distance_fade_min_distance[/code] is greater than [code]distance_fade_max_distance[/code], the behavior will be reversed. The object will start to fade away at [code]distance_fade_max_distance[/code] and will fully disappear once it reaches [code]distance_fade_min_distance[/code].
```

**Review by Calinou — DISMISSED** [review]
Should be good to merge after applying suggestions.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Pull request merged**
